R/caretList.R

```diff
@@ -319,55 +318,91 @@ as.caretList.list <- function(object){
     }
   }
 
-  if(verbose == TRUE){
+  if (verbose == TRUE) {
     pboptions(type = "txt", char = "*")
-  } else if(verbose == FALSE){
+  } else if (verbose == FALSE) {
     pboptions(type = "none")
   }
-  preds <- pbsapply(object, function(x){
+  preds <- pbsapply(object, function(x) {
     type <- x$modelType
-    if (type=="Classification"){
-      if(x$control$classProbs){
-        # Return probability predictions for only one of the classes
-        # as determined by configured default response class level
-        caret::predict.train(x, type="prob", newdata=newdata, ...)[, getBinaryTargetLevel()]
-      } else{
-        caret::predict.train(x, type="raw", newdata=newdata, ...)
+    if (type == "Classification") {
+      if (x$control$classProbs) {
+        caret::predict.train(x, type = "prob", newdata = newdata, ...)
+      } else {
+        caret::predict.train(x, type = "raw", newdata = newdata, ...)
       }
-    } else if(type=="Regression"){
-      caret::predict.train(x, type="raw", newdata=newdata, ...)
-    } else{
+    } else if (type == "Regression") {
+      caret::predict.train(x, type = "raw", newdata = newdata, ...)
+    } else {
       stop(paste("Unknown model type:", type))
     }
   })
-  if(class(preds) != "matrix" & class(preds) != "data.frame"){
-    if(class(preds) == "character" | class(preds) == "factor"){
+
+  if (!inherits(preds, "matrix") && !inherits(preds, "data.frame")) {
+    if (inherits(preds, "character") || inherits(preds, "factor")) {
       preds <- as.character(preds) # drop factorization
     }
     preds <- as.matrix(t(preds))
   }
 
-  if (is.null(names(object))){
+  if (is.null(names(object))) {
     # If the model list used for predictions is not currently named,
     # then exctract the model names from each model individually.
     # Note that this should only be possible when caretList objects
     # are created manually
-    predcols <- sapply(object, extractModelName)
-    colnames(preds) <- make.names(predcols, unique=TRUE)
+    modelnames <- make.names(sapply(object, extractModelName), unique = TRUE)
+  } else {
+    # Otherwise, assign the names of the prediction columns
+    # using the names in the given model list
+    modelnames <- names(object)
+  }
+
+  if (object[[1]]$modelType == "Classification" && object[[1]]$control$classProbs) {
+    # list of data.frames containing probabilities associated to each model
+    probs_model_list <- list()
+    # create indexes to reshape later
+    indexes <- seq_len(nrow(newdata))
+    classes <- levels(object[[1]]$pred$obs)
+
+    for (j in seq_len(ncol(preds))) {
+      modelname <- rep(modelnames[j], length(indexes))
+      class_probabilities <- data.frame(preds[, j])
+      probs_model_list[[modelnames[j]]] <- data.table::data.table(
+        index = indexes, class_probabilities, modelname = modelname
+      )
+    }
+    probs <- data.table::rbindlist(probs_model_list)
```

Comment:
I think you need to add `#' @importFrom table rbindlist` to the roxygen block at the top of this function, so that rbindlist will be in the package namespace.  (I'm not certain of this).

Reply:
I don't think it's necessary, because the `data.table` package is already imported in the `DESCRIPTION` file. According to [Roxygen2 documentation for Imports](https://roxygen2.r-lib.org/articles/namespace.html#imports), it's only necessary to add `@importFrom` when you want to avoid using `package-name::function-name`.

Reply:
got it, thank you!